### PR TITLE
Jonmv/reapply zk changes

### PR DESCRIPTION
**Pull request by jonmv**
@hakonhall please review. @hmusum FYI.

### CODE REVIEW

**Review by hmusum — APPROVED** [review]

**Review by hakonhall — APPROVED** [review]
LGTM but Screwdriver failed

**Comment by jonmv** [comment]
Probably because we try 400 fixed ports, and other unit tests in other modules could grab any of these ...

**Comment by hmusum** [comment]
Ports in unit tests are reserved per module, defined in internal repo

File: zookeeper-server/zookeeper-server-3.6.3/src/test/java/com/yahoo/vespa/zookeper/VespaZooKeeperTest.java [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+    static List<Integer> getPorts(int id) {
+        if (ports.size() < id * 3) {
+            int previousPort;
+            if (ports.isEmpty()) {
+                String[] version = System.getProperty("zk-version").split("\\.");
+                int versionPortOffset = 0;
+                for (String part : version)
+                    versionPortOffset = 32 * (versionPortOffset + Integer.parseInt(part));
+                previousPort = 20000 + versionPortOffset % 30000;
```
Comment (hakonhall) [inline zookeeper-server/zookeeper-server-3.6.3/src/test/java/com/yahoo/vespa/zookeper/VespaZooKeeperTest.java:218]: This 20000 and 30000 has  something to do with the module port range Harald mentioned?  Extract to constants?

**Review by hakonhall — APPROVED** [review]
This doesn't remove the port conflict issue, but it does reduce the chance of it happening right?

File: zookeeper-server/zookeeper-server-3.6.3/src/test/java/com/yahoo/vespa/zookeper/VespaZooKeeperTest.java (hunk 1 above)
Comment (jonmv) [reply zookeeper-server/zookeeper-server-3.6.3/src/test/java/com/yahoo/vespa/zookeper/VespaZooKeeperTest.java:218]: Hmm, they're just picked from thin air, really. Well, those other modules prefer things below 20k, but staying below 50k is just by chance.

**Comment by jonmv** [comment]
Yes, the port conflict may still happen, but there's not really any way to avoid that, when you have to know the port number up front, while not having it bound, i.e., in the interval between choosing it and ZK (re-)binding it. 
The ports are lazily populated to minimise the chances of conflict.

**Comment by jonmv** [comment]
With a Java API for creating ZK servers, we could perhaps pass the sockets? That would eliminate the problem.

**Comment by hakonhall** [comment]
what if we keep an int counter for each zk version, incremented for each new zk version.  This counter could serve as a bit to chose between 2 ranges.  As long as the 2 ranges are within this module's range port allocation should always succeed?

**Comment by jonmv** [comment]
Modules don't really have a static port range. That is outdated. Many tests get random ports from `new ServerSocket(0)`, I think?

**Comment by jonmv** [comment]
And, the ZK-version range stuff is still kept, to also minimise conflict chance.

**Comment by hakonhall** [comment]
> Many tests get random ports from new ServerSocket(0), I think?

I find 4 test files that use it.

**Pull request merged**